staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -1141,6 +1148,14 @@ func (alloc *allocator) checkAvailableCapacity(device deviceWithID) (bool, error
 	return true, nil
 }
 
+func (alloc *allocator) allocatingDeviceForAnyClaim(deviceID DeviceID) bool {
+	return alloc.allocatingDevices[deviceID].Len() > 0
```

Comment:
add check `alloc.allocatingDevices[deviceID] != nil` in all places where `alloc.allocatingDevices[deviceID].something` is called to avoid runtime panic.

Reply:
This type is essentially a map, and `len()`, indexing into, and deleting from a nil map won't actually panic: https://go.dev/play/p/RC5JUhtrIBF. `.Insert()` would, and there's already a check around the one place that's called.

Each place that `.Len` or `.Has` is called is hit in the unit tests for a nil set. The `.Delete` is unreachable with a nil set currently so that isn't covered, but wouldn't panic anyway per the Go playground.